src/test/java/uk/gov/ons/ctp/response/collection/exercise/message/CaseActionEventStatusReceiverIT.java

```diff
@@ -0,0 +1,193 @@
+    Assert.assertEquals(EventDTO.Status.SCHEDULED, initialEvent.getStatus());
+    String eventStatusUpdate =
+        String.format(
+            "{"
+                + "\"collectionExerciseID\": \"%s\","
+                + "\"tag\": \"mps\","
+                + "\"status\":"
+                + "\"COMPLETED\""
+                + "}",
+            collectionExercise.getId());
+    PUBSUBEMULATOR.publishMessage(eventStatusUpdate, PUBSUB_TOPIC);
+    Thread.sleep(5000);
+    Event finalEvent =
+        eventRepository.findOneByCollectionExerciseIdAndTag(collectionExercise.getId(), "mps");
+    Assert.assertEquals(EventDTO.Status.COMPLETED, finalEvent.getStatus());
+  }
+
+  private EventDTO createEventDTO(
+      final CollectionExerciseDTO collectionExercise,
+      final EventService.Tag tag,
+      final Integer daysInFuture) {
+    final EventDTO event = new EventDTO();
+    event.setTimestamp(Date.from(Instant.now().plus(daysInFuture, ChronoUnit.DAYS)));
+    event.setTag(tag.toString());
+    event.setCollectionExerciseId(collectionExercise.getId());
+
+    client.createCollectionExerciseEvent(event);
+    return event;
+  }
+
+  private String loadResourceAsString(Class clazz, String resourceName) throws IOException {
+    InputStream is = clazz.getResourceAsStream(resourceName);
+    StringWriter writer = new StringWriter();
+    IOUtils.copy(is, writer, StandardCharsets.UTF_8.name());
+    return writer.toString();
+  }
+
+  private void stubCollectionInstrumentCount() throws IOException {
+    this.wireMockRule.stubFor(
+        get(urlPathEqualTo("/collection-instrument-api/1.0.2/collectioninstrument/count"))
+            .willReturn(aResponse().withBody("1")));
+  }
+
+  private void createSurveyServiceClassifierStubs() throws IOException {
+    String json =
+        loadResourceAsString(
+            CollectionExerciseEndpointIT.class,
+            "CollectionExerciseEndpointIT.SurveyClassifierDTO.json");
+    this.wireMockRule.stubFor(
+        get(urlPathMatching(
+                "/surveys/([0-9a-f]{8}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{12})"
+                    + "/classifiertypeselectors"))
+            .willReturn(aResponse().withHeader("Content-Type", "application/json").withBody(json)));
+    json =
+        loadResourceAsString(
+            CollectionExerciseEndpointIT.class,
+            "CollectionExerciseEndpointIT.SurveyClassifierTypeDTO.json");
+    this.wireMockRule.stubFor(
+        get(urlPathMatching(
+                "/surveys/([0-9a-f]{8}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{12})"
+                    + "/classifiertypeselectors/"
+                    + "([0-9a-f]{8}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{12})"))
+            .willReturn(aResponse().withHeader("Content-Type", "application/json").withBody(json)));
+  }
+
+  private void stubSurveyServiceBusiness() throws IOException {
+    createSurveyServiceClassifierStubs();
```

Comment:
so you are just rolling up the json errors?

Reply:
yes